ports/esp32/machine_pwm.c

```diff
@@ -152,22 +166,23 @@ STATIC void esp32_pwm_init_helper(esp32_pwm_obj_t *self,
     }
 
     // Maybe change PWM timer
-    int tval = args[ARG_freq].u_int;
-    if (tval != -1) {
-        if (tval != timer_cfg.freq_hz) {
-            if (!set_freq(tval)) {
-                nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError,
-                    "Bad frequency %d", tval));
-            }
-        }
+    int fval = args[ARG_freq].u_int;
+    if (fval != -1 && ledc_set_freq(self->mode, self->timer, fval) != ESP_OK) {
+        nlr_raise(mp_obj_new_exception_msg_varg(&mp_type_ValueError,
+            "Bad frequency %d", fval));
```

Comment:
Please use lower-case at the start of error messages, for consistency with the rest of the code.